packages/vite/src/client/client.ts

```diff
@@ -24,9 +25,9 @@ console.log('[vite] connecting...')
 // use server configuration, then fallback to inference
 const socketProtocol =
   __HMR_PROTOCOL__ || (location.protocol === 'https:' ? 'wss' : 'ws')
-const socketHost = __HMR_PORT__
-  ? `${__HMR_HOSTNAME__ || location.hostname}:${__HMR_PORT__}`
-  : `${__HMR_HOSTNAME__ || location.hostname}`
+const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${
+  __HMR_PORT__ || location.port
+}${__HMR_BASE__}`
```

Comment:
I think we should still support `hmr.port: false`, let's keep it to merge this PR and we can discuss for 3.0 if we want to deprecate the option. I don't see much harm in keeping it now that we already have it. @jeoy, I'm interested to hear your opinion, and feedback in case you could test this PR for your use case.

Reply:
@patak-dev Sure I will try to keep it.

I have to admit I don't fully get the use-case of #6624, but I thought if it is really not needed it might be good to remove now, since it was only released in beta versions :thinking: .
And I agree with https://github.com/vitejs/vite/pull/6624#issuecomment-1021902751 that the port type `number | false` is a bit weird.

Reply:
Oh, you are right. I didn't consider that this option was not still released in a stable version. Good point. @jeoy if this PR covers your use case, I agree with @lukashass here that we should revert #6624 until we find a valid use case for it.

Reply:
Furthermore I just discovered that setting `hmr.port: false` currently doesn't have the desired effect, since `port` will default to `config.server.port` in this line:
https://github.com/vitejs/vite/blob/80afb873311b5c3ef07ae26b92966ab3a49607bf/packages/vite/src/node/plugins/clientInjections.ts#L33